### ISSUE TITLE
Detector physics classes leas memory

### ISSUE THREAD

**Issue by Munken** [adrien-matta/nptool#10]
Most detector physics classes allocate two `XData` with new, but has an empty constructor.

**Comment by Munken** [comment]
Example from AGATA. 

[Allocation](https://github.com/adrien-matta/nptool/blob/NPTool.2.dev/NPLib/Detectors/AGATA/TAGATAPhysics.cxx#L46)
[Empty dtor](https://github.com/adrien-matta/nptool/blob/NPTool.2.dev/NPLib/Detectors/AGATA/TAGATAPhysics.h#L52)

------------------------------

Btw. this also affects the files generated with the wizard.

**Comment by Munken** [comment]
Guess this could also be fixed by having the data classes as value members.

**Comment by adrien-matta** [comment]
Yes it sound like a better plan. I will check if this does not cause any issue with the Branch mechanism, if not I will change it for each detector.

**Comment by adrien-matta** [comment]
After reading carefully the TTree documentation, it seem the user is responsible for deletion of the attached object, so we could delete them. I will make some test later and if it cause no issue change all the detector.

However this is more of an esthetic issue, as the detector classes are instantiate only once in the framework. Still better to do things the right way.

**Comment by Munken** [comment]
Considering your are building are library I find it quite possible that people will link with it. Then they can instantiate as many as they like :wink:
